internal/tiltfile/tiltfile_state.go

```diff
@@ -1000,12 +1001,23 @@ func (s *tiltfileState) k8sResourceForName(name string) (*k8sResource, error) {
 // Otherwise, we'll return the zero value of `s.defaultReg`, which is an empty registry.
 // It has side-effects (a log line) and so should only be called once.
 func (s *tiltfileState) decideRegistry() container.Registry {
-	if s.orchestrator() == model.OrchestratorK8s && !s.localRegistry.Empty() {
+	if s.orchestrator() != model.OrchestratorK8s {
+		return s.defaultReg
+	}
+
+	// Defer lookup of registry until now
+	if s.localRegistry == nil {
```

Comment:
for what it's worth - i'm pretty sure LocalRegistry() is already memoized in the k8sClient itself, so no need to store it as a field on tiltfileState

Reply:
Yeah, I wasn't sure whether it made sense to keep that struct field. I'm going to remove and clean up.